### ISSUE TITLE
`permissions_set_readonly_false` false positive on Windows

### ISSUE THREAD

**Issue by miles170** [rust-lang/rust-clippy#10477]
### Summary

`permissions_set_readonly_false` false positive when running Clippy on Windows.

### Lint Name

permissions_set_readonly_false

### Reproducer

I tried this code on Windows

```rust
let metadata = fs::metadata(path).map_err_context(String::new)?;
let mut perms = metadata.permissions();
perms.set_readonly(false);
```

I saw this happen:

```
error: call to `set_readonly` with argument `false`
   --> src\uu\shred\src/shred.rs:475:9
    |
475 |         perms.set_readonly(false);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: on Unix platforms this results in the file being world writable
    = help: you can set the desired permissions using `PermissionsExt`. For more information, see
            https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#permissions_set_readonly_false
    = note: `-D clippy::permissions-set-readonly-false` implied by `-D warnings`
```

I expected to see no error.

See https://github.com/uutils/coreutils/actions/runs/4383223472/jobs/7673213723

### Version

```text
rustc 1.68.0 (2c8cc3432 2023-03-06)
binary: rustc
commit-hash: 2c8cc343237b8f7d5a3c3703e3a87f2eb2c54a74
commit-date: 2023-03-06
host: x86_64-unknown-linux-gnu
release: 1.68.0
LLVM version: 15.0.6
```

**Comment by Jarcho** [comment]
Checking the repo, it looks like there's nothing in the code to say it's only compiled for Windows. This means that there's no way for clippy to know that there's only one target.

Even with that, I don't think the lint does anything to check if the function call is within a `cfg` block. This would be a good extension for the lint.

**Comment by miles170** [comment]
> ~Checking the repo, it looks like there's nothing in the code to say it's only compiled for Windows. This means that there's no way for clippy to know that there's only one target.~
> 
> Even with that, I don't think the lint does anything to check if the function call is within a `cfg` block. This would be a good extension for the lint.
> 
> Edit: Looks like you're reporting about the pr, not the current master.

Sorry for the incomplete information. I'm indeed reporting about the [PR](https://github.com/uutils/coreutils/pull/4488).

I have updated the code where `perms.set_readonly(false)` is inside the `#[cfg(not(unix))]` block.

**Comment by sylvestre** [comment]
it is this PR:
https://github.com/uutils/coreutils/pull/4488
